sdk/tables/data-tables/src/TableClient.ts

```diff
@@ -38,13 +41,17 @@ import { createPipelineFromOptions, InternalPipelineOptions } from "@azure/core-
 import { logger } from "./logger";
 import { createSpan } from "./utils/tracing";
 import { CanonicalCode } from "@opentelemetry/api";
+import { createBatch, TablesBatch } from "./TablesBatch";
 
 /**
  * A TableClient represents a Client to the Azure Tables service allowing you
  * to perform operations on a single table.
  */
 export class TableClient {
   private table: Table;
+  private url: string;
+  private credential: TablesSharedKeyCredentialLike | undefined;
```

Comment:
Since the batch request is hand crafted, we need access to these 2 properties when sending the batch request later on